var/spack/repos/builtin/packages/hdf/package.py

```diff
@@ -146,10 +146,18 @@ def flag_handler(self, name, flags):
 
         if name == "cflags":
             # https://forum.hdfgroup.org/t/help-building-hdf4-with-clang-error-implicit-declaration-of-function-test-mgr-szip-is-invalid-in-c99/7680
-            if self.spec.satisfies("@:4.2.15 %apple-clang") or self.spec.satisfies("%clang@16:"):
+            if (
+                self.spec.satisfies("%apple-clang")
```

Comment:
this is not a typo I think - there's no version restriction for apple-clang here. We could try setting `-Wno-error=implicit-int` for earlier versions of apple-clang, too. If that works, we can live with one `if` block.